docs/edgeql/overview.rst

```diff
@@ -229,3 +229,74 @@ A query is evaluated recursively using the following procedure:
        the results of the invocations are collected to form a single set.
 
 6. Collect the results of all iterations to obtain the final result set.
+
+
+.. _ref_eql_polymorphic_queries:
+
+Polymorphic Queries
+===================
+
+:index: nested shapes
+
+Sometimes it is useful to have an abstract type as a link target, thus
```

Comment:
Should we also mention unions here?

Reply:
We could _mention_ type unions and link to it without specifically blowing up this example. The type union section has its own polymorphic example already (the only one prior to this and not searchable).

Reply:
I don't like words like "sometimes" in reference documentation.  I'd replace with "A link target can be an abstract type ... "

Reply:
OK